src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpLogExporterHelperExtensions.cs

```diff
@@ -106,29 +109,280 @@ public static OpenTelemetryLoggerOptions AddOtlpExporter(this OpenTelemetryLogge
+        this LoggerProviderBuilder builder,
+        string? name,
+        Action<OtlpExporterOptions, LogRecordExportProcessorOptions>? configureExporterAndProcessor)
+    {
+        var finalOptionsName = name ?? Options.DefaultName;
+
+        builder.ConfigureServices(RegisterOptions);
+
+        return builder.AddProcessor(sp =>
+        {
+            OtlpExporterOptions exporterOptions;
+
+            if (name == null)
+            {
+                // If we are NOT using named options we create a new
+                // instance always. The reason for this is
+                // OtlpExporterOptions is shared by all signals. Without a
+                // name, delegates for all signals will mix together. See:
+                // https://github.com/open-telemetry/opentelemetry-dotnet/issues/4043
+                exporterOptions = sp.GetRequiredService<IOptionsFactory<OtlpExporterOptions>>().Create(finalOptionsName);
+            }
+            else
+            {
+                // When using named options we can properly utilize Options
+                // API to create or reuse an instance.
+                exporterOptions = sp.GetRequiredService<IOptionsMonitor<OtlpExporterOptions>>().Get(finalOptionsName);
+            }
+
+            var processorOptions = sp.GetRequiredService<IOptionsMonitor<LogRecordExportProcessorOptions>>().Get(finalOptionsName);
+
+            // Configuration delegate is executed inline.
+            configureExporterAndProcessor?.Invoke(exporterOptions, processorOptions);
+
+            // Note: Not using finalOptionsName here for SdkLimitOptions.
+            // There should only be one provider for a given service
+            // collection so SdkLimitOptions is treated as a single default
+            // instance.
+            var sdkLimitOptions = sp.GetRequiredService<IOptionsMonitor<SdkLimitOptions>>().CurrentValue;
+
+            return BuildOtlpLogExporter(
+                sp,
+                exporterOptions,
+                processorOptions,
+                sdkLimitOptions,
+                sp.GetRequiredService<IOptionsMonitor<ExperimentalOptions>>().Get(finalOptionsName));
+        });
+    }
+
     internal static BaseProcessor<LogRecord> BuildOtlpLogExporter(
         IServiceProvider sp,
         OtlpExporterOptions exporterOptions,
         LogRecordExportProcessorOptions processorOptions,
         Func<BaseExporter<LogRecord>, BaseExporter<LogRecord>>? configureExporterInstance = null)
     {
-        if (sp == null)
-        {
-            throw new ArgumentNullException(nameof(sp));
-        }
-
-        Debug.Assert(exporterOptions != null, "exporterOptions was null");
-        Debug.Assert(processorOptions != null, "processorOptions was null");
+        Debug.Assert(sp != null, "sp was null");
 
-        var config = sp.GetRequiredService<IConfiguration>();
+        var config = sp!.GetRequiredService<IConfiguration>();
 
         var sdkLimitOptions = new SdkLimitOptions(config);
```

Comment:
The newly added `AddOtlpLogExporter` methods are registering OptionsFactory for `SdkLimitOptions` and `ExperimentalOptions`.

Shouldn't we be using `IOptionsFactory<TOptions>.Create` method to create these options?

Reply:
The `SdkLimitOptions` and `ExperimentalOptions` will be registered into the `IOptionsFactory<T>` which takes IConfiguration as one of the parameters:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/e3759a1e0ea6e23b80cad33a206795fda8e240ff/src/Shared/Options/ConfigurationExtensions.cs#L137C1-L145C12
So the options can be retrieved when constructing new option instances with `IConfiguration`.

Reply:
My question is instead of using the regular ctor for `SdkLimitOptions` and `ExperimentalOptions`, shouldn't we be using either: `IOptionsFactory<TOptions>.Create` or `IOptionsMonitor<TOptions>.Get` to get these options. Shouldn't we be utilizing the `OptionsFactory` that we are now registering in this PR for these two options?

Reply:
It isn't as important to use Options API with `SdkLimitOptions` and `ExperimentalOptions` because they are `internal` but I went ahead and refactored it so it is nice and consistent now.